File: 0152-maximum-product-subarray/0152-maximum-product-subarray.java
class Solution {
    public int maxProduct(int[] nums) {
        int n = nums.length;
        int maxP = nums[0];
        for(int i = 0; i<n; i++){
            int currP = 1;
            for(int j = i; j<n; j++){
                currP *= nums[j];
                maxP = Math.max(maxP, currP);
            }
        }
        return maxP;
    }
}

File: 0152-maximum-product-subarray/0152-maximum-product-subarray.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int n = nums.size();
        int p = nums[0];
        for(int i = 0; i<n; i++){
            int product = nums[i];
            p = max(product,p);
            for(int j = i+1; j < n; j++){
                product *= nums[j];
                if(p < product){
                    p = product;
                }
            }
        }
        return p;
    }
};